src/main/java/it/gov/pagopa/reward/service/reward/RewardCalculatorMediatorServiceImpl.java

```diff
@@ -76,6 +78,7 @@ private Mono<RewardTransactionDTO> evaluate(TransactionDTO trx, List<String> ini
                 .mapNotNull(userCounters -> evaluateInitiativesBudgetAndRules(trx, initiatives, userCounters))
                 .flatMap(counters2rewardedTrx -> {
                     userInitiativeCountersUpdateService.update(counters2rewardedTrx.getFirst(), counters2rewardedTrx.getSecond());
+                    transactionProcessedService.saveTransactionProcessed(trx);
```

Comment:
maybe better store counters2rewardedTrx.getSecond() inside an explicit variable
```suggestion
                    transactionProcessedService.saveTransactionProcessed(counters2rewardedTrx.getSecond());
```

MoreOver, the save operation should return a Mono and the "userInitiativeCountersRepository.save...." should be call after it